### ISSUE TITLE
aws-eks: allow passing secret to `cluster.addHelmChart`

### ISSUE THREAD

**Issue by trondhindenes** [aws/aws-cdk#16476]
it would be good if CDK would resolve references to SecretsManager for `values` passed to `addHelmChart`

### Use Case
I'm trying to do this: read a secret, apply a helm chart with values referencing that secret. Code:
```python
    secret = secretsmanager.Secret.from_secret_name_v2(scope, 'linkerDSecret', '/kubernetescluster/linkerd')
    cluster.add_helm_chart(
        id='linkerd2',
        chart='linkerd2',
        repository='https://helm.linkerd.io/stable',
        create_namespace=True,
        release='linkerd',
        values={
            'identityTrustAnchorsPEM': secret.secret_value_from_json(key='identityTrustAnchorsPEM'),
            'identity': {
                'issuer': {
                    'crtExpiry': secret.secret_value_from_json(key='identityissuerCertExpiry'),
                    'tls': {
                        'crtPEM': secret.secret_value_from_json(key='identityissuertlscrtPEM'),
                        'keyPEM': secret.secret_value_from_json(key='identityissuertlskeyPEM'),
                    }
                }
            }
        }
    )
```
However, this seems to not be supported - secrets are not resolved. It would be good if this was supported, so that we have a way of sending secrets to Helm charts.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by otaviomacedo** [comment]
I am unassigning and marking this issue as `p2`, which means that we are unable to work on this immediately.

We use +1s to help prioritize our work, and are happy to revaluate this issue based on community feedback. You can reach out to the cdk.dev community on [Slack](https://join.slack.com/t/cdk-dev/shared_invite/zt-mso6p56d-qJp7SOTBvMaQuDrx7R2wHg) to solicit support for reprioritization.

**Comment by markus7811** [comment]
I like this issue, but it's somehow similar to the following ones:

https://github.com/aws/aws-cdk/issues/15853
https://github.com/aws/aws-cdk/issues/9815

The main problem is, that cloudformation can't handle this

> Dynamic references for secure values, such as secretsmanager, aren't currently supported in custom resources.
[cloudformation documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-secretsmanager)

So i see no 'native' way of implementing this.

A workaround could be extending the kubectl/helm handler with [vals](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-secretsmanager). I use it a lot in combination with [helmfile]https://github.com/roboll/helmfile and it offers access for a lot backends. 

Wht do you think about this solution? 

The question would be: how do we handle the iam permissions that we have to attach to the kubectl/helm handler...

**Comment by trondhindenes** [comment]
(I thought about publishing this to the constructs catalog, but haven't quite finished making it open-source-friendly yet)

**Comment by iliapolo** [comment]
Yeah this is an annoying one. We don't have an action plan yet but its on our radar now.

**Comment by vechorko** [comment]
any road map or estimation time for implementation it? it's really major feature in use helm charts in  big infrastructure to pass passwords and secret date from another stacks and services

**Comment by akmalharith** [comment]
@trondhindenes I wonder if you are using kubectlprovider by cdk, aren’t the manifests being applied actually printing the secrets in the Lambda logs. How are you suppressing them?

**Comment by stephanpelikan** [comment]
We would also need that feature, passing DB credentials. I will try to use external-secrets instead.

**Comment by jwan6060** [comment]
Any roadmap to implement this?

**Comment by shaundclements** [comment]
looking forward to secret resolution for both addHelm and addManifest
